frontend/alert/components/Timeline/index.tsx

```diff
@@ -0,0 +1,287 @@
+    display: grid;
+    grid-template-columns: [start] 22% [date] 20% [time] 35% [end];
+    justify-items: center;
+    width: 100%;
+    align-items: start;
+    justify-content: center;
+    margin-right: 1rem;
+`;
+
+const getMonthDay = (timeString) => {
+    let d = new Date(timeString);
+    return d.toLocaleDateString("en-US", { month: "numeric", day: "numeric" });
+};
+
+const formatData = (courseData) => {
+    /* Convert the course status data into an array of
+     * length 3 arrays with the status data object, the current status at each created_at data point
+     * and if the date is different
+     */
+    let formattedData = courseData.reduce((ans, item, index) => {
+        const sameDate =
+            index == 0 ||
+            getMonthDay(courseData[index]["created_at"]) ==
+                getMonthDay(courseData[index - 1]["created_at"]);
+
+        if (item["old_status"] == "C" && item["new_status"] == "O") {
+            return ans.concat([[item, "opened", sameDate]]);
+        } else {
+            return ans.concat([[item, "closed", sameDate]]);
+        }
+    }, []);
+
+    return formattedData;
+};
+
+interface TimelineProps {
+    courseCode: string | null;
+    setTimeline: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+const Timeline = ({ courseCode, setTimeline }: TimelineProps) => {
+    const [courseStatusData, setCourseStatusData] = useState([]);
+    const [loaded, setLoaded] = useState(false);
+    const [scrollTimeline, setScrollTimeline] = useState(false);
+
+    const close = !courseCode;
+
+    //hook that detects when a user is scrolling a component & when they stop scrolling
+    const useScroll = (ref) => {
+        useEffect(() => {
+            var timer;
+            const handleScroll = (e) => {
+                if (!scrollTimeline) {
+                    setScrollTimeline(true);
+                    clearTimeout(timer);
+                    timer = setTimeout(function () {
+                        setScrollTimeline(false);
+                    }, 1000);
+                }
+            };
+
+            ref.current.addEventListener("scroll", handleScroll, true);
+        }, [ref]);
+
+        return ref;
+    };
+
+    useEffect(() => {
+        if (!courseCode) {
+            return;
+        }
+
+        //loading course status update data
+        setLoaded(false);
+
+        fetch(`/api/base/statusupdate/${courseCode}/`).then((res) =>
+            res.json().then((courseResult) => {
+                //sort data by when it was created from earliest to latest
+                courseResult.sort((a, b) =>
+                    a.created_at < b.created_at ? 1 : -1
+                );
+
+                setCourseStatusData(formatData(courseResult));
+
+                setLoaded(true);
+            })
+        );
+    }, [courseCode]);
+
+    return (
+        <AlertHistoryContainer
+            close={close}
+            ref={useScroll(
+                useOnClickOutside(
```

Comment:
the `useScroll` and `useOnClickOutside` hooks should get bound to `const` calues before the return statement and then have those values used inline instead of calling the hooks inline. Since this always returns the same hooks it doesn't expliclty break one of the [rules of hooks](https://reactjs.org/docs/hooks-rules.html), but in case this component gets changed going forward to do some conditional rendering, better to have the hooks specified in the top part of the function.